### PR TITLE
Ensure an `ObjectPoolProvider` is registered

### PR DESCRIPTION
**Pull request by dougbu**
- e.g. take advantage of aspnet/HttpAbstractions#561 fix wherever cookies are used

### CODE REVIEW

File: test/Microsoft.AspNetCore.Hosting.Tests/WebHostBuilderTests.cs [hunk 1]
```diff
@@ -16,6 +16,7 @@
 using Microsoft.Extensions.PlatformAbstractions;
 using System.Reflection;
 using Xunit;
+using Microsoft.Extensions.ObjectPool;
```
Comment (Tratcher) [inline test/Microsoft.AspNetCore.Hosting.Tests/WebHostBuilderTests.cs]: sort

File: src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs [hunk 2]
```diff
@@ -201,6 +202,9 @@ private IServiceCollection BuildHostingServices()
             // Conjure up a RequestServices
             services.AddTransient<IStartupFilter, AutoRequestServicesStartupFilter>();
 
+            // Ensure object pooling is available everywhere.
+            services.TryAddSingleton<ObjectPoolProvider, DefaultObjectPoolProvider>();
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs]: Try?

Comment (dougbu) [inline src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs]: Why _not_ allow users to override the provider? In any case, this method uses both Try and unconditional; I went with Try.

File: test/Microsoft.AspNetCore.Hosting.Tests/WebHostBuilderTests.cs (hunk 1 above)
Comment (dougbu) [inline test/Microsoft.AspNetCore.Hosting.Tests/WebHostBuilderTests.cs]: Ooops. (Likely wasn't placed correctly because the list was unsorted before.)

**Comment by dougbu** [comment]
:up::date:

**Comment by davidfowl** [comment]
:shipit:

**Comment by dougbu** [comment]
b06a844

**Pull request closed without merging**